src/data/preview.ts

```diff
@@ -1,21 +1,25 @@
 import { UnsubscribeFunc } from "home-assistant-js-websocket";
 import { HomeAssistant } from "../types";
 
-export interface ThresholdPreview {
+export interface GenericPreview {
   state: string;
   attributes: Record<string, any>;
 }
 
-export const subscribePreviewThreshold = (
+export const subscribePreviewGeneric = (
   hass: HomeAssistant,
+  domain: string,
   flow_id: string,
   flow_type: "config_flow" | "options_flow",
   user_input: Record<string, any>,
-  callback: (preview: ThresholdPreview) => void
+  callback: (preview: GenericPreview) => void
 ): Promise<UnsubscribeFunc> =>
   hass.connection.subscribeMessage(callback, {
-    type: "threshold/start_preview",
+    type: `${domain}/start_preview`,
     flow_id,
     flow_type,
     user_input,
   });
+
+export const previewModule = (domain: string): string =>
+  ["template"].includes(domain) ? domain : "generic";
```

Comment:
This is very much hidden that template still has its own frontend component. Should we add a const HAS_CUSTOM_PREVIEW = [ "template" ] at the top to make this clearer?

Reply:
Makes sense. Or could also try/catch to load the domain module first, and fallback to generic if it doesn't exist, and then not need to maintain a list?

Reply:
I would prefer the list. If it fails to load the domain module for network issues or during development of new config flow previews, it would be less confusing.